### ISSUE TITLE
Cylc processes with high CPU usage

### ISSUE THREAD

**Issue by penguian** [ACCESS-NRI/accessdev-Trac-archive#265]
`resolution_fixed`   |    by mrd599@nci.org.au

___

Top sometimes shows cylc processes with CPU usage consistently around 50%. The cause of this seems to be that they are continually writing files to cylc-run/SUITE/state, e.g. in this case at a frequency of about 1.5 s. Note that cylc keeps only the most recent 10 files

```
  69956 Mar  1 14:32 state.20160301T033212.313480Z
  69956 Mar  1 14:32 state.20160301T033213.906387Z
  69956 Mar  1 14:32 state.20160301T033215.450920Z
  69956 Mar  1 14:32 state.20160301T033217.007761Z
  69956 Mar  1 14:32 state.20160301T033218.547831Z
  69956 Mar  1 14:32 state.20160301T033220.227907Z
  69956 Mar  1 14:32 state.20160301T033221.708751Z
  69956 Mar  1 14:32 state.20160301T033223.677265Z
  69956 Mar  1 14:32 state.20160301T033225.011035Z
  69956 Mar  1 14:32 state.20160301T033226.531301Z
```

These files are all identical apart from the internal timestamp. In suites behaving normally these files are only written when something changes (e.g. a task finishes). Here however nothing seems to be changing at this high frequency. The log files don't show anything strange.

I've only seen this behaviour in complex NWP suites with lots of tasks. The high CPU usage can also come and go while the suite runs.

Seems more likely to be a cylc bug rather than anything to do with our local configuration?

https://groups.google.com/forum/#!topic/cylc/RyTpSfar61A

___
_Issue migrated from [trac:265](https://accessdev.nci.org.au/trac/265) at 2024-01-31 18:23:32 +1100_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _commented_


___
It's a bug https://github.com/cylc/cylc/issues/1744

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _changed status from `new` to `assigned`_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _set owner to `mrd599`_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _commented_


___
Still sometimes see high CPU usage (~30%) even with 6.9.1 suites, though there's no longer the constant writing of state files.

It seems to occur in APS3 trial suites when there are internally queued tasks, though it seems independent of the actual number of queued tasks. When there are no queued tasks the CPU usage drops back to normal (1-2%).

Simpler suites with queued tasks don't show the problem.

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _commented_


___
Another cylc bug https://github.com/cylc/cylc/issues/1787

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _changed status from `assigned` to `closed`_

**Comment by penguian** [comment]
_@martin.dix@anu.edu.au_ _commented_


___
Fixed in cylc 6.10.0.
